src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/StripEmbeddedLibraries.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using System;
+using System.Linq;
+using Xamarin.Android.Tasks;
+
+namespace MonoDroid.Tuner
+{
+	public class StripEmbeddedLibraries : BaseStep
+	{
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (!Annotations.HasAction (assembly))
+				return;
+
+			var action = Annotations.GetAction (assembly);
+			if (action == AssemblyAction.Skip || action == AssemblyAction.Delete)
+				return;
+
+			var fileName = assembly.Name.Name + ".dll";
+			if (MonoAndroidHelper.IsFrameworkAssembly (fileName) && !MonoAndroidHelper.FrameworkEmbeddedJarLookupTargets.Contains (fileName))
+				return;
+
+			bool assembly_modified = false;
+			foreach (var mod in assembly.Modules) {
```

Comment:
This should also remove resources which *start* with `__AndroidEnvironment__`, which are also embedded into assemblies.

Reply:
👍 I guess we weren't doing that at all before???

Reply:
This is added now